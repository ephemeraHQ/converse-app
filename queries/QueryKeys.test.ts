import {
  QueryKeys,
  groupDescriptionQueryKey,
  groupFirstMessageQueryKey,
  groupMembersQueryKey,
  groupMessagesQueryKey,
  groupNameQueryKey,
  groupPendingMessagesQueryKey,
  groupPermissionsQueryKey,
  groupPhotoQueryKey,
  groupPinnedFrameQueryKey,
  groupQueryKey,
  groupsQueryKey,
} from "./QueryKeys";

describe("QueryKeys", () => {
  it("should match the snapshot", () => {
    expect(QueryKeys).toMatchSnapshot();
  });
});

describe("Query Key Functions", () => {
  const account = "testAccount";
  const topic = "testTopic";

  it("groupsQueryKey should return the correct array", () => {
    const result = groupsQueryKey(account);
    expect(result).toEqual([QueryKeys.GROUPS, account]);
  });

  it("groupQueryKey should return the correct array", () => {
    const result = groupQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP, account, topic]);
  });

  it("groupMessagesQueryKey should return the correct array", () => {
    const result = groupMessagesQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_MESSAGES, account, topic]);
  });

  it("groupFirstMessageQueryKey should return the correct array", () => {
    const result = groupFirstMessageQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_FIRST_MESSAGE, account, topic]);
  });

  it("groupPendingMessagesQueryKey should return the correct array", () => {
    const result = groupPendingMessagesQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_PENDING_MESSAGES, account, topic]);
  });

  it("groupMembersQueryKey should return the correct array", () => {
    const result = groupMembersQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_MEMBERS, account, topic]);
  });

  it("groupNameQueryKey should return the correct array", () => {
    const result = groupNameQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_NAME, account, topic]);
  });

  it("groupDescriptionQueryKey should return the correct array", () => {
    const result = groupDescriptionQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_DESCRIPTION, account, topic]);
  });

  it("groupPhotoQueryKey should return the correct array", () => {
    const result = groupPhotoQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_PHOTO, account, topic]);
  });

  it("groupPinnedFrameQueryKey should return the correct array", () => {
    const result = groupPinnedFrameQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.PINNED_FRAME, account, topic]);
  });

  it("groupPermissionsQueryKey should return the correct array", () => {
    const result = groupPermissionsQueryKey(account, topic);
    expect(result).toEqual([QueryKeys.GROUP_PERMISSIONS, account, topic]);
  });
});
