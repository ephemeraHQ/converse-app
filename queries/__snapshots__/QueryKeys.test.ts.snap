// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryKeys should match the snapshot 1`] = `
{
  "ADDED_BY": "addedBy",
  "GROUP": "group",
  "GROUPS": "groupsv2",
  "GROUP_ACTIVE": "groupActive",
  "GROUP_CONSENT": "groupConsent",
  "GROUP_DESCRIPTION": "groupDescription",
  "GROUP_FIRST_MESSAGE": "groupFirstMessage",
  "GROUP_INVITE": "groupInvite",
  "GROUP_JOIN_REQUEST": "groupJoinRequest",
  "GROUP_MEMBERS": "groupMembersv2",
  "GROUP_MESSAGES": "groupMessages",
  "GROUP_NAME": "groupName",
  "GROUP_PENDING_MESSAGES": "groupPendingMessages",
  "GROUP_PERMISSIONS": "groupPermissions",
  "GROUP_PHOTO": "groupPhoto",
  "PENDING_JOIN_REQUESTS": "pendingJoinRequests",
  "PINNED_FRAME": "pinnedFrame",
}
`;
